docs/3.1/docs/app-dev/explicit-contract-disclosure.rst

```diff
@@ -264,3 +264,23 @@ These last two steps are executed using the new Daml Script functions supporting
 
 .. note:: For an example using Java bindings for client applications, see the
   `Java Bindings StockExchange example project <https://github.com/digital-asset/ex-java-bindings/blob/f474ae83976b0ad197e2fabfce9842fb9b3de907/StockExchange/README.rst>`_.
+
+Safeguards
+------------------------------------------
+Explicit Contract Disclosure usage should always be accompanied by on-ledger contracts to ensure workflows executed based on the disclosed contract's contents conform to on-ledger agreements between the stakeholders (or trusted parties) involved.
+
+In the above example, ``IOU`` contracts between **Buyer** and **Issuer** (a trusted party on the ledger) ensure that the exercising of the ``Offer_Accept`` choice using disclosed contract results in a contractually agreed-upon outcome.
+
+This works by performing several safety checks in the ``IOU_Transfer`` choice within the ``IOU`` contract, which is called from the ``Offer_Accept`` choice.
+
+The ``IOU`` contract provides several safeguards in the ``Offer_Accept`` workflow:
+
+- **Buyer** exercising the ``Offer_Accept`` choice is defined on the ``IOU`` agreement.
+- Creation of an IOU with an same amount for the **Seller** happens atomically with a deduction of the same amount from the **Buyer'** IOU.
+- **Buyer** cannot be deducted and **Seller** cannot receive more than the stipulated value on the ``IOU`` contract.
+
+By ensuring the **Buyer** party expected to execute the ``Offer_Accept`` choice, a trusted **Issuer** party and required terms of execution are clearly defined on an on-ledger multi-signatory ``IOU`` contract, the **Seller** can disclose the ``Offer`` contract and the **Buyer** can execute the ``Offer_Accept`` choice on the disclosed ``Offer`` contract knowing workflow safety is ensured.
+
+If disclosed contracts contain malicious data or are maliciously executed on, the safeguards will prevent unexpected outcomes outside the bounds of the agreement stated on the ``IOU`` contract from resolving.
```

Comment:
```suggestion
If disclosed contracts contain malicious data or are maliciously executed on, the safeguards prevent unexpected outcomes. These safeguards ensure that any issues remain within the bounds of the agreement stated in the IOU contract and do not lead to unresolved consequences.
```

Reply:
I realized the 2nd part of my sentence is already mentioned earlier here: https://github.com/digital-asset/docs.daml.com/pull/774#discussion_r1705534062

I've removed it and kept just the first part.